### ISSUE TITLE
SLS rendering errors could be more helpful

### ISSUE THREAD

**Issue by jdshewey** [saltstack/salt#44501]
### Description of Issue/Question
When referencing a pillar that does not exist, an error is thrown such as:

Rendering SLS 'base:katello.server' failed: Jinja variable 'str object' has no attribute 'gpg_key' this error could be more helpful by including the line number of the SLS or the ID Declaration. This is particularly useful in large files where a variable like gpg_key could be referenced often and it is difficult to find the instance that was fat-fingered.

**Comment by gtmanfred** [comment]
Yeah, that seems reasonable, I am marking this as a feature request.

Thanks,
Daniel

**Comment by jdshewey** [comment]
bump

**Comment by helephant** [comment]
I'm also currently wishing for line numbers :)

I'm getting the somewhat baffling: "Trying to populate non string value into a string for variable ${deep:4}. Please make sure the value of the property is a string."

**Comment by driskell** [comment]
I spend most of my hours finding errors in big SLS files. This is not so much a feature request but a big requirement.

**Comment by jdshewey** [comment]
bump
